test/Oscoin/Test/P2P/Disco.hs

```diff
@@ -0,0 +1,234 @@
+module Oscoin.Test.P2P.Disco
+    ( tests
+    -- shut up, weeder!
+    -- , prop_multicast
+    -- , prop_staticPeers
+    -- , prop_dnsSD
+    )
+where
+
+import           Oscoin.Prelude hiding (retry)
+
+import           Oscoin.P2P.Disco (Options(..), withDisco)
+import qualified Oscoin.P2P.Disco.MDns as MDns
+import           Oscoin.P2P.Types (Network, readNetwork, readNodeAddr)
+
+import           Control.Retry
+import qualified Data.Set as Set
+import qualified Network.DNS as DNS
+import           Network.Socket
+                 ( AddrInfo(..)
+                 , AddrInfoFlag(..)
+                 , HostAddress
+                 , HostAddress6
+                 , PortNumber
+                 , SockAddr(..)
+                 , SocketType(Datagram)
+                 , bind
+                 , close
+                 , defaultHints
+                 , defaultProtocol
+                 , getAddrInfo
+                 , socket
+                 , socketPort
+                 , tupleToHostAddress
+                 , tupleToHostAddress6
+                 )
+import           Network.Socket.ByteString (recvFrom, sendAllTo)
+
+import           Hedgehog
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+import           Test.Tasty (TestTree, testGroup)
+import           Test.Tasty.Hedgehog (testProperty)
+
+tests :: TestTree
+tests = testGroup "Disco"
+    [ testProperty "prop_multicast"   prop_multicast
+    , testProperty "prop_staticPeers" prop_staticPeers
+    , testProperty "prop_dnsSD"       prop_dnsSD
+    ]
+
+{-
+props :: IO Bool
```

Comment:
Needed anymore?

Reply:
Yeah, I took the liberty to expose those from all the test modules below `Oscoin.Test.P2P`, and teach `weeder` to shut up about unused exports once and for all. The type is `IO Bool`, which makes it very convenient to run just this test group in ghci.

/cc @cloudhead @geigerzaehler